### PR TITLE
Feature/intl duration format

### LINKED ISSUE #21: Replace momentjs
Momentjs is sunsetting. These are the alternatives: https://github.com/you-dont-need/You-Dont-Need-Momentjs/blob/master/README.md

My favourite: day-fns

### PR DESCRIPTION
**Pull request by mreiche**
Added support for [Intl.RelativeTimeFormat](https://v8.dev/features/intl-relativetimeformat).
May fix #21

### CODE REVIEW

File: tests/value-converters/intl-duration-format-value-converter.test.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        expectedFormat: "yesterday, 1 hour ago"
+    },
+    {
+        input: <DateTimeComponents>{
+            seconds: 4100,
+        },
+        optionId: "short",
+        locale: "en",
+        expectedFormat: "in 1 hr. in 8 min. in 20 sec."
+    },
```
Comment (Zsar) [inline tests/value-converters/intl-duration-format-value-converter.test.ts:106]: Note that `ms` components will be silently dropped. E.g.
```typescript
{
  input: <DateTimeComponents>{ ms: -1, },
  optionId: "short",
  locale: "en",
  expectedFormat: "1 ms. ago"
},
```
will fail, producing the empty string instead.

File: src/utils/time.ts [hunk 2]
```diff
@@ -61,33 +60,76 @@ export function normalizeTime(date: Date, components: TimeComponents) {
...
-    // YEAR=MONTH*12,
+    DAY=HOUR*24,
+    WEEK=DAY*7,
+    MONTH=DAY*30,
+    QUARTER=MONTH*3,
+    YEAR=DAY*365,
 }
 
-export function toMilliseconds(components: TimeComponents) {
+export function toMilliseconds(components: DateTimeComponents) {
```
Comment (Zsar) [inline src/utils/time.ts:79]: It's likely not practically relevant, but should we not at least give a hint via JSDoc on the interfaces, that we cannot handle arithmetics on, say
```typescript
<DateTimeComponents> {
  ms: Number.MAX_VALUE,
  seconds: Number.MAX_VALUE,
}
```
or any other case where the sum of fields overflows?

Comment (Zsar) [reply src/utils/time.ts:79]: I am not entirely current with TypeScript features, but I believe we cannot feasibly restrict the domain of our interfaces to values we can process as-expected.

Given the presumably exotic nature of this error class, I'd not, on the other hand, want to introduce actual runtime checks to watch out for it.

So documentation of the _actually intended_ domain seems to be the sensible (middle) ground.

**Review by Zsar — APPROVED** [review]

File: tests/value-converters/intl-duration-format-value-converter.test.ts (hunk 1 above)
Comment (mreiche) [reply tests/value-converters/intl-duration-format-value-converter.test.ts:106]: Milliseconds are not supported by `IntlDurationFormatValueConverter`.

File: tests/value-converters/intl-duration-format-value-converter.test.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    {
+        input: <DateTimeComponents>{
+            days: 3,
+            hours: 4,
+            minutes: 5,
+            seconds: 6
+        },
+        optionId: null,
+        expectedFormat: "in 3 days, in 4 hours, in 5 minutes, in 6 seconds"
```
[the remark this answers is not in the document]
Comment (mreiche) [reply tests/value-converters/intl-duration-format-value-converter.test.ts:48]: There is no proper solution right now. I'd like to work on that in another PR.

File: src/utils/time.ts (hunk 2 above)
Comment (mreiche) [reply src/utils/time.ts:79]: I don't want to taint my code for except unusual edge cases like Number overflows.
```
Number.MAX_VALUE*2 // Infinity
```

File: tests/value-converters/intl-duration-format-value-converter.test.ts (hunk 1 above)
Comment (Zsar) [reply tests/value-converters/intl-duration-format-value-converter.test.ts:106]: Fair enough. Should probably add a sentence in the readme: "Milliseconds will be dropped", or some such.

Comment (mreiche) [reply tests/value-converters/intl-duration-format-value-converter.test.ts:106]: Thats not true, because milliseconds could overflow to seconds.

File: src/utils/time.ts (hunk 2 above)
Comment (Zsar) [reply src/utils/time.ts:79]: Yes, me neither. Would be more code for the edge case than for the normal behaviour.

That's why I recommend putting an information as JSDoc onto the three interfaces: "Values should not sum to Number.MAX_VALUE or Number.MIN_VALUE".

That way it's just an RTFM and no longer our concern.

File: tests/value-converters/intl-duration-format-value-converter.test.ts (hunk 1 above)
Comment (Zsar) [reply tests/value-converters/intl-duration-format-value-converter.test.ts:106]: Mmh. True. Badly formulated. "Remaining milliseconds will be dropped"?

File: src/utils/time.ts (hunk 2 above)
Comment (mreiche) [reply src/utils/time.ts:79]: `Number.MAX_VALUE` would be `1.7976931348623157e+305` seconds. I dont know if the universe will still exist then.

Comment (Zsar) [reply src/utils/time.ts:79]: No, true, it will never show up as result of a meaningful calculation. It might be a consecutive fault, though. But very well.

**Comment by mreiche** [comment]
I cannot merge.
